hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -366,20 +370,27 @@ static void validateTableIsIceberg(Table table, String fullName) {
    * The decision is made like this:
    * <ol>
    * <li>Table property value {@link TableProperties#ENGINE_HIVE_ENABLED}
-   * <li>If the table property is not set then check the hive-site.xml property value
+   * <li>If the table property is not set then check hive metastore TABLE_PARAMS value
    * {@link ConfigProperties#ENGINE_HIVE_ENABLED}
+   * <li>If the table property is not set and hive metastore TABLE_PARAMS is not set
+   * then check the hive-site.xml property value{@link ConfigProperties#ENGINE_HIVE_ENABLED}
    * <li>If none of the above is enabled then use the default value {@link TableProperties#ENGINE_HIVE_ENABLED_DEFAULT}
    * </ol>
    * @param metadata Table metadata to use
+   * @param parameters Hive table parameters to use
    * @param conf The hive configuration to use
    * @return if the hive engine related values should be enabled or not
    */
-  private static boolean hiveEngineEnabled(TableMetadata metadata, Configuration conf) {
+  private static boolean hiveEngineEnabled(TableMetadata metadata, Map<String, String> parameters, Configuration conf) {
```

Comment:
Load the hiveEngine Enabled priorities are as follows(a>b>c):
a. Table properties
b. Hive table parameters
c. System configuration

Reply:
@zhangdove: We have a related [discussion](https://lists.apache.org/thread.html/rfc9d82fd5f76cad4b8b7baea1bf00c0f4351feed4692c1a2ffa30788%40%3Cdev.iceberg.apache.org%3E) on the iceberg dev list which I think is not closed yet.

We are trying to find a place for the different properties, and the end result should define also how the ENGINE_HIVE_ENABLED property should be handled. Maybe this should be an "Iceberg table property" which should be stored on the Iceberg table itself.

Feel free to add your voice to the discussion!

Reply:
@pvary I've seen some discussion(it’s great work), but based on that, could we do the following things here?

1. modify Hive related [docs](https://github.com/apache/iceberg/blob/master/site/docs/hive.md#hive-configuration), at least in terms of usage? (disable ENGINE_HIVE_ENABLED, and prompt the user to use [another way](https://github.com/apache/iceberg/blob/master/site/docs/hive.md#table-property-configuration) )

2. save the value of `ENGINE_HIVE_ENABLED` in the Configuration into the "Iceberg table property"

Of course,if there is a better solution, I am very interested in how to deal with hive engine enable priorities, and close this pr.

Thanks.

Reply:
@zhangdove: To be honest, I am not sure I have understand everything you have proposed. There are several alternative ways to enable Hive configuration on the tables ATM:

1. Global configuration: Here we have to set `iceberg.engine.hive.enabled=true` in the hive-site.xml or other global config
2. Local configuration: Every client making changes to the table should set `iceberg.engine.hive.enabled=true` in the configuration they are using
3. Table configuration: The table property (`engine.hive.enabled`) should be set directly on the table.

Would this be appropriate for you?

Thanks,
Peter

Reply:
@pvary Thanks for your reply!
Let me try another question:
Because the data cannot be read when using the case of #1831, do we need to update the Hive document [hive-configuration](https://github.com/apache/iceberg/blob/master/site/docs/hive.md#hive-configuration)?

Reply:
@zhangdove: Would this work if the following spark session is used?
```
    val spark = SparkSession
      .builder()
      .master("local[2]")
      .appName("IcebergAPI")
      .config("spark.sql.catalog.hive_prod", "org.apache.iceberg.spark.SparkCatalog")
      .config("spark.sql.catalog.hive_prod.type", "hive")
      .config("spark.sql.catalog.hive_prod.uri", "thrift://localhost:9083")
      .config("iceberg.engine.hive.enabled", "true")
      .enableHiveSupport()
      .getOrCreate()
```
Notice, I have added `.config("iceberg.engine.hive.enabled", "true")`

Reply:
@pvary I guess and have been verified in the Configuration of SparkSession add `.config("iceberg.engine.hive.enabled", "true")`, it's work.

Thanks.